### PR TITLE
[split build] move periodic split builds into own concurrency group

### PR DESCRIPTION
**Pull request by PaliC**
To avoid nightly workflows cancelling each other

### CODE REVIEW

**Review by clee2000 — APPROVED** [review]

**Review by huydhn — APPROVED** [review]
LGTM! So, just to confirm we want to run split build with `ciflow/periodic` on PR but on all trunk commits (push to main, not periodic)

**Review by malfet — APPROVED** [review]

File: .github/scripts/generate_ci_workflows.py [hunk 1]
```diff
@@ -70,6 +70,8 @@ def __post_init__(self) -> None:
             )
         else:
             self.build_environment = f"{self.os}-binary-{self.package_type}"
+        if self.use_split_build:
+            self.build_environment += "-split"  # added to distinguish concurrency groups
```
Comment (malfet) [inline .github/scripts/generate_ci_workflows.py:74]:
```suggestion
        if self.use_split_build:
            # added to distinguish concurrency groups
            self.build_environment += "-split"
```

**Pull request closed without merging**